solr/core/src/java/org/apache/solr/update/processor/DistributedStandaloneUpdateProcessor.java

```diff
@@ -56,8 +59,15 @@ public void doDeleteByQuery(DeleteUpdateCommand cmd) throws IOException {
     super.doDeleteByQuery(cmd, Collections.emptyList(), null);
   }
 
+  @Override
+  protected void postProcessDeleteByQuery(DeleteUpdateCommand cmd,
+                                          List<SolrCmdDistributor.Node> replicas, DocCollection coll) throws IOException {
+    // no op
+  }
+
   @Override
   boolean isLeader(UpdateCommand cmd) {
+    updateCommand = cmd;
```

Comment:
ouch this piece is ugly.  a method like "is..." (or get...)  should not have side effects.

Reply:
since this is a package private method, perhaps it should be renamed to setupRequest(like the protected methods) just that this once takes an update command( Distributed#setupRequest(UpdateCommand cmd) ). It should be stated in the java doc that the method returns whether this node is the leader for the supplied update command(cmd).
WDYT?

Reply:
+1 to setupRequest name.  I don't think it needs to return "isLeader"; it should set this and return void.

Reply:
Perhaps we should have another isLeader method that returns a boolean, since the current method is used by other classes in the package to setup the request and determine whether the core is the leader for the specified update command.
I'll work on a patch ASAP.